tests/Query/BuilderTest.php

```diff
@@ -433,6 +434,62 @@ function (Builder $builder) {
                 ->orWhereNotBetween('id', collect([3, 4])),
         ];
 
+        yield 'where like' => [
+            ['find' => [['name' => new Regex('^acme$', '')], []]],
```

Comment:
If the `like` pattern is not using any special operators, does it make sense to produce a regex instead of a direct value match? I realize `ilike` would still need a regex for case-insensitivity.

Both this and the `where like escape` test below could avoid using a regex object.

If you agree, feel free to defer this to another JIRA ticket, since it's probably just an optimization.

Reply:
Since like is now case-insensitive, this optimization is no longer possible.

Reply:
Agreed. I was writing that suggestion alongside my other suggestion that we _not_ break BC in case we went in either direction. Feel free to resolve after reading.